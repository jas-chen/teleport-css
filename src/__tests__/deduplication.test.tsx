import { expect, test } from 'vitest';
import { renderToStaticMarkup } from 'react-dom/server';
import fnv1a from '@sindresorhus/fnv1a';
import { create } from '..';
import pretty from 'pretty';

function hashFn(value: string) {
  return fnv1a(value, { size: 64 }).toString(36).slice(0, 8);
}

const { styled, keyframes } = create({
  hashFn,
});

test('deduplication', () => {
  const spin = keyframes(() => ({
    from: { transform: 'rotate(0deg)' },
    to: { transform: 'rotate(360deg)' },
  }));

  const Button = styled('button', () => ({
    animation: `${spin} 2s ease infinite`,
    animationName: `${spin}`,
  }));

  const PrimaryButton = styled(Button, () => ({
    animation: `${spin} 1s ease infinite`,
  }));

  expect(pretty(renderToStaticMarkup(<PrimaryButton />))).toMatchSnapshot();
});
