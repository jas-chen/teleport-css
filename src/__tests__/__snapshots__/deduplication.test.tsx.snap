// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`deduplication 1`] = `
"<style data-precedence="0" data-href="x_2bn4sldq">
  @keyframes x_2bn4sldq {
    from {
      transform: rotate(0deg)
    }

    to {
      transform: rotate(360deg)
    }
  }
</style>
<style data-precedence="1" data-href="x_3iq8y4v01">
  .x_3iq8y4v01 {
    animation-name: x_2bn4sldq
  }
</style>
<style data-precedence="2" data-href="x_15ga9oq42">
  .x_15ga9oq42 {
    animation: x_2bn4sldq 1s ease infinite
  }
</style><button class="x_3iq8y4v01 x_15ga9oq42"></button>"
`;
